mmgen/models/architectures/biggan/modules.py

```diff
@@ -0,0 +1,416 @@
+        # Here in_channels of BigGANGenResBlock equal to output_dim of ccbn
+        self.bn1 = BigGANConditionBN(
+            in_channels,
+            dim_after_concat,
+            sn_eps=sn_eps,
+            label_input=label_input,
+            with_spectral_norm=with_spectral_norm,
+            auto_sync_bn=auto_sync_bn)
+        self.bn2 = BigGANConditionBN(
+            out_channels,
+            dim_after_concat,
+            sn_eps=sn_eps,
+            label_input=label_input,
+            with_spectral_norm=with_spectral_norm,
+            auto_sync_bn=auto_sync_bn)
+
+        self.conv1 = SNConvModule(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=3,
+            stride=1,
+            padding=1,
+            conv_cfg=conv_cfg,
+            act_cfg=None,
+            with_spectral_norm=with_spectral_norm,
+            spectral_norm_cfg=dict(eps=sn_eps))
+
+        self.conv2 = SNConvModule(
+            in_channels=out_channels,
+            out_channels=out_channels,
+            kernel_size=3,
+            stride=1,
+            padding=1,
+            conv_cfg=conv_cfg,
+            act_cfg=None,
+            with_spectral_norm=with_spectral_norm,
+            spectral_norm_cfg=dict(eps=sn_eps))
+
+    def forward(self, x, y):
+        # TODO:
+        """[summary]
+
+        Args:
+            x ([type]): [description]
+            y ([type]): [description]
+
+        Returns:
+            [type]: [description]
+        """
+        x0 = self.bn1(x, y)
+        x0 = self.activation(x0)
+        if self.with_upsample:
+            x0 = self.upsample_layer(x0)
+            x = self.upsample_layer(x)
+        x0 = self.conv1(x0)
+        x0 = self.bn2(x0, y)
+        x0 = self.activation(x0)
+        x0 = self.conv2(x0)
+        if self.learnable_sc:
+            x = self.shortcut(x)
+        return x0 + x
+
+
+@MODULES.register_module()
+class BigGANConditionBN(nn.Module):
+    """[summary]
+
+    Args:
+        num_features ([type]): [description]
+        input_dim ([type]): [description]
+        bn_eps ([type], optional): [description]. Defaults to 1e-5.
+        sn_eps ([type], optional): [description]. Defaults to 1e-6.
+        momentum (float, optional): [description]. Defaults to 0.1.
+        label_input (bool, optional): [description]. Defaults to False.
+        with_spectral_norm (bool, optional): [description]. Defaults to
+            True.
+        auto_sync_bn (bool, optional): [description]. Defaults to True.
+    """
+
+    def __init__(self,
+                 num_features,
+                 input_dim,
+                 bn_eps=1e-5,
+                 sn_eps=1e-6,
+                 momentum=0.1,
+                 label_input=False,
+                 with_spectral_norm=True,
+                 auto_sync_bn=True):
+        super().__init__()
+        assert num_features > 0 and input_dim > 0
+        # Prepare gain and bias layers
+        if not label_input:
```

Comment:
Please specify the design of the label in this part.  What's the relation between `input_dim` and `num_features`. You should also consider changing the variable name in this module.